### PR TITLE
Better top level read me.

### PR DESCRIPTION
**Pull request by bianpengyuan**
Signed-off-by: Pengyuan Bian <bianpengyuan@google.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,39 @@
 # Istio Ecosystem Wasm Extensions
+
 [![Test Status][test-badge]][test-link]
 
 This repository contains several canonical Wasm extensions, which intend to demonstrate:
-* Development pattern of Wasm extension.
+
+* Wasm extension development pattern.
```
Comment (kyessenov) [inline README.md:7]: _patterns_

File: README.md [hunk 2]
```diff
@@ -1,26 +1,39 @@
...
 
-# Extensions
-* [Basic auth](/extensions/basic_auth/)
-* [C++ scaffold](/extensions/scaffold/)
-* [gRPC access logging](./extension/grpc_loggig)
-* [Local rate limit](/extensions/local_rate_limit/)
-* [Zig scaffold](/extensions/zig_demo/)
+## Extensions
+
+* *[Basic auth](/extensions/basic_auth/)* enforces basic auth based on request host, path, and methods. From this extension, you can find how to perform local auth decision based on headers and local reply, as well as JSON configuration string parsing and base64 decoding.
```
Comment (kyessenov) [inline README.md:12]: _In this extension_

File: README.md [hunk 3]
```diff
@@ -1,26 +1,39 @@
...
+  
+* *[C++ scaffold](/extensions/scaffold/)* provides an empty C++ extension, which can be used as a start point to write a C++ Wasm extension.
```
Comment (kyessenov) [inline README.md:14]: _starting point_

File: README.md [hunk 4]
```diff
@@ -1,26 +1,39 @@
...
+  
+* *[gRPC access logging](./extensions/grpc_logging)* makes a logging request to a gRPC service with various kinds of request and workload attributes. From this extension, you can find how to perform asynchronous telemetry reporting, fetch various request attributes and proxy properties, use protobuf and make gRPC callout.
+
+* *JWT based routing ([WIP](https://github.com/istio-ecosystem/wasm-extensions/issues/16))* reads JWT token information from Envoy dynamic metadata written by JWT auth filter, update host header accordingly, and trigger routing recomputation. From this extension, you can find how to read dynamic metadata, manipulate headers, and affect request routing.
```
Comment (kyessenov) [inline README.md:18]: _In this extension_

File: README.md [hunk 5]
```diff
@@ -1,26 +1,39 @@
...
+
+* *[Local rate limit](/extensions/local_rate_limit/)* applies a token bucket rate limit to incoming requests. Each request processed by the filter utilizes a single token, and if no tokens are available, the request will be denied. From this extension you can find how to share data across all plugin VMs and deny request with local reply.
```
Comment (kyessenov) [inline README.md:20]: is denied, In this extension

File: README.md [hunk 6]
```diff
@@ -1,26 +1,39 @@
...
 
-# Guides
+* *Open Policy Agent client ([WIP](https://github.com/istio-ecosystem/wasm-extensions/pull/54))* makes HTTP callout to a Open Policy Agent (OPA) server and based on OPA server response make decision to allow or deny an incoming request. A result cache is also included to avoid expensive callout on every request. From this extension, you can find how to perform HTTP callout, and asynchronously continue or stop an incoming request based on the response of HTTP call. You will also find how to record stats, which can be scraped in the same way as Istio standard metrics.
```
Comment (kyessenov) [inline README.md:22]: an Open Policy Agent... make decision -> decides whether

File: README.md [hunk 7]
```diff
@@ -1,26 +1,39 @@
...
+
+* *[Zig scaffold](/extensions/zig_demo/)* provides an empty [Zig](https://ziglang.org/) extension, which can be used as a start point to write a Zig Wasm extension.
```
Comment (kyessenov) [inline README.md:24]: a starting point

File: README.md [hunk 8]
```diff
@@ -1,26 +1,39 @@
...
+
+## Development Guides
+
+### Write a Wasm Extension with C++
 
-## C++
 * [Development set up](doc/development-setup.md)
-* [Write, test, deploy, and maintain a Wasm extension](./doc/write-a-wasm-extension-with-cpp.md)
+* [Write, test, deploy, and maintain a C++ Wasm extension](./doc/write-a-wasm-extension-with-cpp.md)
+* [Write unit test with C++ Wasm extension](./doc/write-cpp-unit-test.md)
```
Comment (kyessenov) [inline README.md:32]: You can add a link to https://github.com/istio-ecosystem/wasm-extensions/wiki/Wasm-Development-Tips&Tricks or convert it into a doc.

**Review by kyessenov — APPROVED** [review]

File: README.md (hunk 8 above)
Comment (bianpengyuan) [reply README.md:32]: Yeah I'd like to convert that into a doc, but latter. Added it here.

**Pull request merged**